src/pydvl/influence/torch/operator.py

```diff
@@ -471,3 +472,101 @@ def _apply_to_mat(self, mat: torch.Tensor) -> torch.Tensor:
     @property
     def input_size(self) -> int:
         return self.batch_operation.input_size
+
+
+class LowRankOperator(TensorOperator):
+    r"""
+    Given a low rank representation of a matrix
+
+    $$ A = V D V^T$$
+
+    with a diagonal matrix $D$ and an optional regularization parameter $\lambda$,
+    computes
+
+    $$ (V D V^T+\lambda I)^{-1}b$$.
+
+    Depending on the value of the `exact` flag, the inverse action is computed exactly
+    using the [Sherman–Morrison–Woodbury formula]
+    (https://en.wikipedia.org/wiki/Woodbury_matrix_identity). If `exact` is set to
+    `False`, the inverse action is approximated by
+
+    $$ V^T(D+\lambda I)^{-1}Vb$$
+
+    Args:
+    """
+
+    def __init__(
+        self,
+        low_rank_representation: LowRankProductRepresentation,
+        regularization: float,
+        exact: bool = True,
+    ):
+
+        if exact and (regularization is None or regularization <= 0):
+            raise ValueError("regularization must be positive when exact=True")
+        elif regularization is not None and regularization < 0:
+            raise ValueError("regularization must be non-negative")
+
+        self._regularization = regularization
+        self._exact = exact
+        self._low_rank_representation = low_rank_representation
+
+    @property
+    def exact(self):
+        return self._exact
+
+    @property
+    def regularization(self):
+        return self._regularization
+
+    @regularization.setter
+    def regularization(self, value: float):
+        if value < 0:
+            raise ValueError("regularization must be non-negative")
+        self._regularization = value
+
+    @property
+    def device(self):
+        return self._low_rank_representation.device
+
+    @property
+    def dtype(self):
+        return self._low_rank_representation.dtype
+
+    def to(self, device: torch.device):
+        self._low_rank_representation = self._low_rank_representation.to(device)
+        return self
+
+    def _apply_to_vec(self, vec: torch.Tensor) -> torch.Tensor:
+
+        if vec.ndim == 1:
+            return self._apply_to_mat(vec.unsqueeze(0)).squeeze()
+
+        return self._apply_to_mat(vec)
+
+    def _apply_to_mat(self, mat: torch.Tensor) -> torch.Tensor:
+
+        regularized_eigen_vals = self._low_rank_representation.eigen_vals.clone()
+
+        if self.regularization is not None:
+            regularized_eigen_vals += self.regularization
+
+        proj_rhs = self._low_rank_representation.projections.t() @ mat.t()
+        inverse_regularized_eigenvalues = 1.0 / regularized_eigen_vals
+        result = self._low_rank_representation.projections @ (
+            proj_rhs * inverse_regularized_eigenvalues.unsqueeze(-1)
+        )
+
+        if self._exact:
+            result += (
+                1.0
+                / self.regularization
+                * (mat.t() - self._low_rank_representation.projections @ proj_rhs)
+            )
```

Comment:
I'm usually pro long names but here I found it a bit hard to relate the math snippet in the docstring to the code. If I understand correctly:
- `self._low_rank_presentation.eigen_vals` = `D`
- `self._low_rank_presentation.projections` = `V`

I think it would be easier to read if you assign variables that make the actual computations look more like the math snippet.

Reply:
Agree, introduced auxiliary variables `V` and `D`. I think it reads much clearer now.